### PR TITLE
Remove deprecated code and deprecate a few more things in io.fits

### PR DESCRIPTION
**Pull request by saimn**
- Deprecate _ExtensionHDU and _NonstandardExtHDU
- Remove special handling of TCTYP TCUNI TCRPX TCRVL TCDLT TRPOS (#7157)
- Rename TableHDU.update to .update_header

### Description

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
Looks good! Nice to get rid of some old cruft.

File: docs/changes/io.fits/15396.api.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Deprecate ``_ExtensionHDU`` and ``_NonstandardExtHDU`` (use ``ExtensionHDU`` or
+``NonstandardExtHDU`` instead).
+
+Remove special handling of TCTYP TCUNI TCRPX TCRVL TCDLT TRPOS (#7157)
```
Comment (pllim) [inline docs/changes/io.fits/15396.api.rst:4]:
```suggestion
Remove special handling of TCTYP TCUNI TCRPX TCRVL TCDLT TRPOS (#7157).
```

**Comment by pllim** [comment]
You are very brave to deprecate/remove stuff in FITS...

**Review by astrofrog — APPROVED** [review]
Looks good to me!

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**